starknet-replay/src/storage/rpc/mod.rs

```diff
@@ -75,58 +112,30 @@ pub struct RpcStorage {
     endpoint: Url,
 
     /// The client field sends RPC calls.
-    client: Client,
+    client: JsonRpcClient<HttpTransport>,
 }
 impl RpcStorage {
     /// Constructs a new `RpcStorage`.
     ///
     /// # Arguments
     ///
     /// - `endpoint`: The Url of the Starknet RPC node.
-    ///
-    /// # Errors
-    ///
-    /// Returns [`Err`] if [`jsonrpc::minreq_http::MinreqHttpTransport`] can't
-    /// be created.
-    pub fn new(endpoint: Url) -> Result<Self, DatabaseError> {
-        let t = MinreqHttpTransport::builder()
-            .url(endpoint.to_string().as_str())?
-            .build();
-
-        let client = Client::with_transport(t);
-        Ok(RpcStorage { endpoint, client })
-    }
-
-    /// This function makes an RPC call and returns the response.
-    ///
-    /// # Arguments
-    ///
-    /// - `method`: The method of the RPC calls
-    /// - `args`: The parameters of the RPC call. `None` if there are no
-    ///   parameters.
-    ///
-    /// # Errors
-    ///
-    /// Returns [`Err`] if the request fails.
-    fn send_request(
-        &self,
-        method: &str,
-        args: Option<&RawValue>,
-    ) -> Result<Response, DatabaseError> {
-        let request = self.client.build_request(method, args);
-        tracing::info!("jsonrpc request {request:?}");
-        Ok(self.client.send_request(request)?)
+    #[must_use]
+    pub fn new(endpoint: Url) -> Self {
+        let client = JsonRpcClient::new(HttpTransport::new(endpoint.clone()));
+        RpcStorage { endpoint, client }
     }
 
     /// This function queries the number of the most recent Starknet block.
     ///
     /// # Errors
     ///
     /// Returns [`Err`] if the request fails.
-    pub fn starknet_block_number(&self) -> Result<BlockNumber, DatabaseError> {
-        let response = self.send_request("starknet_blockNumber", None)?;
-        let result: u64 = response.result()?;
-        Ok(BlockNumber::new(result))
+    #[allow(clippy::missing_panics_doc)] // To avoid false positives caused by OK() statement.
```

Comment:
This doesn't make sense to me. How does the `Ok` statement create false positives for panics?